File: signing-consentor-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__SigningConsentor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ConsentAttributes;
import se.curity.identityserver.sdk.oauth.consent.SigningConsentorResult;
import se.curity.identityserver.sdk.oauth.consent.signing.SigningConsentor;

public final class ${pluginName}SigningConsentor implements SigningConsentor
{
    private final ${pluginName}ConsentorConfig _configuration;

    public ${pluginName}SigningConsentor(${pluginName}ConsentorConfig configuration)
    {
        _configuration = configuration;

        // See examples of consentors on GitHub: https://github.com/search?q=topic%3Aconsentor+org%3Acurityio
    }

    @Override
    @Deprecated
    public SigningConsentorResult start(String textToDisplay, String objectToSign)
    {
        throw new UnsupportedOperationException();
    }

    @Override
    public SigningConsentorResult start(String textToDisplay, String objectToSign, ConsentAttributes consentAttributes)
    {
        throw new UnsupportedOperationException();
    }

    @Override
    public SigningConsentorResult poll()
    {
        throw new UnsupportedOperationException();
    }

    @Override
    public void cancel()
    {
        throw new UnsupportedOperationException();
    }
}

File: kotlin-signing-consentor-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__SigningConsentor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.ConsentAttributes
import se.curity.identityserver.sdk.oauth.consent.SigningConsentorResult
import se.curity.identityserver.sdk.oauth.consent.signing.SigningConsentor

class ${pluginName}SigningConsentor(private val _configuration: ${pluginName}ConsentorConfig): SigningConsentor
{
    @Deprecated("Use the implementation with consentAttributes instead")
    override fun start(textToDisplay: String, objectToSign: String): SigningConsentorResult
    {
        TODO()
    }

    override fun start(textToDisplay: String, objectToSign: String, consentAttributes: ConsentAttributes): SigningConsentorResult
    {
        TODO()
    }

    override fun poll(): SigningConsentorResult
    {
        TODO()
    }

    override fun cancel()
    {
        TODO()
    }
}
